File: external_apps/mobile_ios/mobile_ios/Core/Config/RemoteConfigManager.swift
//
//  RemoteConfigManager.swift
//  mobile_ios
//
//  Created for DevOps & Operations
//

import Foundation
import Combine

class RemoteConfigManager: ObservableObject {
    static let shared = RemoteConfigManager()
    
    @Published var config: [String: Any] = [:]
    
    // Default config values
    private let defaultAPITimeout: Int64 = 30000
    private let defaultMaxRetries: Int = 3
    private let defaultCacheTTL: Int64 = 3600
    
    private init() {
        config = getDefaultConfig()
    }
    
    /// Get default configuration
    private func getDefaultConfig() -> [String: Any] {
        return [
            "api_timeout": defaultAPITimeout,
            "max_retries": defaultMaxRetries,
            "cache_ttl": defaultCacheTTL,
            "enable_analytics": true,
            "enable_crash_reporting": true,
            "min_app_version": "1.0.0",
            "force_update_version": "1.0.0",
            "maintenance_mode": false,
            "maintenance_message": "App is under maintenance"
        ]
    }
    
    /// Fetch remote config from server
    func fetchConfig() async {
        // TODO: Fetch from Firebase Remote Config
        // let remoteConfig = RemoteConfig.remoteConfig()
        // try await remoteConfig.fetchAndActivate()
        
        // For now, use default config
        config = getDefaultConfig()
    }
    
    /// Get string value
    func getString(_ key: String, defaultValue: String = "") -> String {
        return config[key] as? String ?? defaultValue
    }
    
    /// Get boolean value
    func getBoolean(_ key: String, defaultValue: Bool = false) -> Bool {
        return config[key] as? Bool ?? defaultValue
    }
    
    /// Get integer value
    func getInt(_ key: String, defaultValue: Int = 0) -> Int {
        if let value = config[key] as? Int {
            return value
        } else if let value = config[key] as? Int64 {
            return Int(value)
        }
        return defaultValue
    }
    
    /// Get double value
    func getDouble(_ key: String, defaultValue: Double = 0.0) -> Double {
        if let value = config[key] as? Double {
            return value
        } else if let value = config[key] as? Float {
            return Double(value)
        }
        return defaultValue
    }
    
    /// Get all config values
    func getAllConfig() -> [String: Any] {
        return config
    }
    
    /// Check if maintenance mode is enabled
    func isMaintenanceMode() -> Bool {
        return getBoolean("maintenance_mode", defaultValue: false)
    }
    
    /// Get maintenance message
    func getMaintenanceMessage() -> String {
        return getString("maintenance_message", defaultValue: "App is under maintenance")
    }
    
    /// Get API timeout
    func getApiTimeout() -> Int {
        return getInt("api_timeout", defaultValue: Int(defaultAPITimeout))
    }
    
    /// Get max retries
    func getMaxRetries() -> Int {
        return getInt("max_retries", defaultValue: defaultMaxRetries)
    }
}

File: external_apps/mobile_android/app/src/main/java/com/example/mobile_android/data/config/RemoteConfigManager.kt
package com.example.mobile_android.data.config

import android.content.Context
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Remote Config Manager
 * 
 * Note: Integrate with Firebase Remote Config
 * 
 * Add to build.gradle:
 * implementation("com.google.firebase:firebase-config-ktx")
 */

@Singleton
class RemoteConfigManager @Inject constructor(
    @ApplicationContext private val context: Context
) {
    
    private val _config = MutableStateFlow<Map<String, Any>>(getDefaultConfig())
    val config: StateFlow<Map<String, Any>> = _config
    
    companion object {
        // Default config values
        private const val DEFAULT_API_TIMEOUT = 30000L
        private const val DEFAULT_MAX_RETRIES = 3
        private const val DEFAULT_CACHE_TTL = 3600L
    }
    
    /**
     * Get default configuration
     */
    private fun getDefaultConfig(): Map<String, Any> {
        return mapOf(
            "api_timeout" to DEFAULT_API_TIMEOUT,
            "max_retries" to DEFAULT_MAX_RETRIES,
            "cache_ttl" to DEFAULT_CACHE_TTL,
            "enable_analytics" to true,
            "enable_crash_reporting" to true,
            "min_app_version" to "1.0.0",
            "force_update_version" to "1.0.0",
            "maintenance_mode" to false,
            "maintenance_message" to "App is under maintenance"
        )
    }
    
    /**
     * Fetch remote config from server
     */
    suspend fun fetchConfig() {
        // TODO: Fetch from Firebase Remote Config
        // val remoteConfig = Firebase.remoteConfig
        // remoteConfig.fetchAndActivate().await()
        
        // For now, use default config
        _config.value = getDefaultConfig()
    }
    
    /**
     * Get string value
     */
    fun getString(key: String, defaultValue: String = ""): String {
        return _config.value[key] as? String ?: defaultValue
    }
    
    /**
     * Get boolean value
     */
    fun getBoolean(key: String, defaultValue: Boolean = false): Boolean {
        return _config.value[key] as? Boolean ?: defaultValue
    }
    
    /**
     * Get long value
     */
    fun getLong(key: String, defaultValue: Long = 0L): Long {
        return when (val value = _config.value[key]) {
            is Long -> value
            is Int -> value.toLong()
            else -> defaultValue
        }
    }
    
    /**
     * Get double value
     */
    fun getDouble(key: String, defaultValue: Double = 0.0): Double {
        return when (val value = _config.value[key]) {
            is Double -> value
            is Float -> value.toDouble()
            else -> defaultValue
        }
    }
    
    /**
     * Get all config values
     */
    fun getAllConfig(): Map<String, Any> = _config.value
    
    /**
     * Check if maintenance mode is enabled
     */
    fun isMaintenanceMode(): Boolean {
        return getBoolean("maintenance_mode", false)
    }
    
    /**
     * Get maintenance message
     */
    fun getMaintenanceMessage(): String {
        return getString("maintenance_message", "App is under maintenance")
    }
    
    /**
     * Get API timeout
     */
    fun getApiTimeout(): Long {
        return getLong("api_timeout", DEFAULT_API_TIMEOUT)
    }
    
    /**
     * Get max retries
     */
    fun getMaxRetries(): Int {
        return getLong("max_retries", DEFAULT_MAX_RETRIES.toLong()).toInt()
    }
}
